code/modules/admin/debug.dm

```diff
@@ -363,6 +363,12 @@ var/global/debug_messages = 0
 					if (match)
 						listargs += match
 
+			if ("ref")
+				var/input = input("Enter ref:") as null|text
+				var/target = locate(input)
+				if (!target) target = locate("\[[input]\]")
```

Comment:
Looks sus, this doesn't runtime?

Reply:
if it doesn't have brackets it evaluates out to null as the ref is invalid

example from datumvars.dm of similar

```dm
			var/theref = input("What ref?") as null|text
			if(theref)
				var/thing = locate(theref)
				if(!thing)
					thing = locate("\[[theref]\]")
				if(!thing)
					boutput(src, "<span class='alert'>Bad ref or couldn't find that thing. Drats.</span>")
					return
```

Reply:
surprised invalid refs properly error